### PR TITLE
ECHO-1529: Added changelog version-0.13

### PR DESCRIPTION
**Pull request by Ashes17b**

### CODE REVIEW

File: changelogs/0.13.md [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+- Improved synchronization in echorand plugin
+- Echorand, async dispatching message to active agreement
+- Added limit for x86 contract size in the `create_contract_operation`
+- Handling zero-balances and zero-block at chain start
+- Contract results getters
+- Optimize `interval_tree_selector::get_points`
+
+## Refactoring code with Static Analyzer Code
+
+## Changed data structures
```
Comment (viktor32) [inline changelogs/0.13.md:69]: Probably we should create files with these structures and leave here only a link to files

File: api-reference/echo-objects/objects-order.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# Objects order
+
+| Objects | id |
+| :--- | :--- |
+|null_object_type|0|
+|base_object_type|1|
+|account_object_type|2|
```
Comment (kstdl) [inline api-reference/echo-objects/objects-order.md]: id should be id_type(e.g.: 1.2.x for accounts)

File: api-reference/echo-objects/dynamic-global-property-object.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+Maintains global state information (committee_member list, current fees)
+
+This is an implementation detail. The values here are calculated during normal chain operations and reflect the current values of global blockchain properties.
+
+```cpp
+class dynamic_global_property_object : public abstract_object<dynamic_global_property_object>
+{
+    public:
+        static const uint8_t space_id = implementation_ids;
```
Comment (kstdl) [inline api-reference/echo-objects/dynamic-global-property-object.md]: remove unneeded data from structs

File: changelogs/0.13.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
+# [0.13.0]
+
+## Added
+- Bitcoin sidechain
+    - Operation `sidechain_btc_approve_aggregate_operation`
```
Comment (kstdl) [inline changelogs/0.13.md]: there should be link to op

File: changelogs/0.13.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
+# [0.13.0]
+
+## Added
```
Comment (kstdl) [inline changelogs/0.13.md:3]: describe for what are needed all added or add links to another page with full description

File: changelogs/0.13.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+- Fields `reciever`, `amount` to [block_reward_operation](/api-reference/echo-operations/block-reward.md#block_reward_operation)
+- Field `x86_64_maximum_contract_size` to `chain_parameters`
+- Database API
+    - New argument `type` (`` or `eth` or `btc`) for [get_account_withdrawals](/api-reference/echo-node-api/database-api.md#get_account_withdrawals-account), [get_account_deposits](/api-reference/echo-node-api/database-api.md#get_account_deposits-account)
+    - Method `get_block_rewards`
+- Added tests for x86_64 call of contract using `chain_interface`
+
+## Changed
+- Bitcoin sidechain
+	- Field `committee_member_ids_in_script` in `btc_address_object`, `btc_intermediate_deposit_object`, `btc_aggregating_object` is `map<account_id_type, echo::sidechain::btc::public_key>` now
```
Comment (kstdl) [inline changelogs/0.13.md]: looks like mess

File: changelogs/0.13.md [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+	- Rename method `get_active_committee_set` to `get_active_committee_pub_keys_map`
+- Field `task` in `registration_api` is `optional<registration_task>` now
+- Wallet API
+    - Rename `get_btc_addresses` to [get_btc_addresses](/api-reference/echo-wallet-api/README.md#get_btc_addresses-account)
+- Now frozen balances are taken into account, changed `get_balance` in `interval_tree_selector_impl`
+- Maximum number active committee members from `1001` to `19`
+- Database API
+    - Method `get_account_deposits` return type `variants` now
```
Comment (kstdl) [inline changelogs/0.13.md]: what could be in variants? maybe links should be added

File: changelogs/0.13.md [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+- Field `next_available_vote_id` from `global_property_object`
+- Mechanism increase registration fee
+- Field `maximum_committee_count` from `chain_parameters`
+- Field `committee_member` from `committee_frozen_balance_deposit_operation`
+- Field `owner_approvals_to_remove` from `proposal_update_operation`
+- Fields `owner_approvals_to_add`, `owner_approvals_to_remove` from `approval_delta`
+
+## Fixed
+- Fallback signatures verification
+- Issue block_reward_operation for every account
```
Comment (kstdl) [inline changelogs/0.13.md]: Block reward are accumulated and issues with block_reward_operation during maintenance

File: changelogs/0.13.md [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+- Committee_member_create_operation at zero deposit
+- Bugs in `chain_interface`
+- Incorrect withdraw in Ethereum sidechain
+- Bug with updating ethereum address of a committee member
+- Sidechain called twice in the emulator
+- Signatures check in method `aggregate_btc` from `btc_sidechain`
+- SIGSEGV during messages processing
+- Calling `echorand::on_transaction`
```
Comment (kstdl) [inline changelogs/0.13.md]: remove

File: changelogs/0.13.md [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+- Problem with block application because of invalid state of websocket
+- Deprecated-copy error with gcc 9.x
+- Build macOS with unused-parameter as error
+- `Registrer_account` example in wallet-API
```
Comment (kstdl) [inline changelogs/0.13.md]: remove. changed comment of method for doxygen

File: changelogs/0.13.md [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+
+## Improved
+- Updated Ethereum sidechain to reflect the absence of empty blocks
+- Implement CPFP logic and related functionality
+- Separate `on_maintanance` and `on_committee_update` signals in Database
+- Improved synchronization in echorand plugin
+- Echorand, async dispatching message to active agreement
+- Added limit for x86 contract size in the `create_contract_operation`
+- Handling zero-balances and zero-block at chain start
+- Contract results getters
```
Comment (kstdl) [inline changelogs/0.13.md]: Contract results getters are optimized

File: changelogs/0.13.md [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+- `Registrer_account` example in wallet-API
+- Fallback message generation
+
+## Improved
+- Updated Ethereum sidechain to reflect the absence of empty blocks
+- Implement CPFP logic and related functionality
+- Separate `on_maintanance` and `on_committee_update` signals in Database
+- Improved synchronization in echorand plugin
+- Echorand, async dispatching message to active agreement
+- Added limit for x86 contract size in the `create_contract_operation`
```
Comment (kstdl) [inline changelogs/0.13.md]: mentioned in `added` section. remove

File: changelogs/0.13.md [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+- Build macOS with unused-parameter as error
+- `Registrer_account` example in wallet-API
+- Fallback message generation
+
+## Improved
+- Updated Ethereum sidechain to reflect the absence of empty blocks
+- Implement CPFP logic and related functionality
+- Separate `on_maintanance` and `on_committee_update` signals in Database
+- Improved synchronization in echorand plugin
+- Echorand, async dispatching message to active agreement
```
Comment (kstdl) [inline changelogs/0.13.md]: Echorand messages are dispatched asyncly

**Review by kstdl — CHANGES_REQUESTED** [review]

**Pull request merged**